dpctl-capi/source/dpctl_sycl_device_manager.cpp

```diff
@@ -168,6 +168,32 @@ DPCTLDeviceMgr_GetDevices(int device_identifier)
     return wrap(Devices);
 }
 
+__dpctl_give const char *
+DPCTLDeviceMgr_GetDeviceInfoStr(__dpctl_keep const DPCTLSyclDeviceRef DRef)
+{
+    char *cstr_info = nullptr;
+    auto D = unwrap(DRef);
+    if (D) {
+        try {
+            auto infostr = get_device_info_str(*D);
+            auto cstr_len = infostr.length() + 1;
+            cstr_info = new char[cstr_len];
+#ifdef _WIN32
+            strncpy_s(cstr_info, cstr_len, infostr.c_str(), cstr_len);
+#else
+            std::strncpy(cstr_info, infostr.c_str(), cstr_len);
+#endif
```

Comment:
We have code snippets such as this sprinkled around the codebase. I think we should encapsulate them into a callable function and use that.

Reply:
agreed. I will make the change.

Reply:
Done in a58afe9c11e2a9bf8352af85fbde1ce9502f8c2d